src/vs/workbench/contrib/terminal/browser/media/terminal.css

```diff
@@ -81,8 +87,23 @@
 	z-index: 31;
 }
 
-.monaco-workbench .simple-find-part-wrapper {
+.monaco-workbench .simple-find-part-wrapper.terminal-find {
 	z-index: 33 !important;
+	overflow: unset;
+	padding-right: 80px;
+}
+
+.terminal-find .simple-find-part {
+	width: 280px;
+	max-width: 280px;
+	min-width: 280px;
+}
+
+.terminal-find .matchesCount {
```

Comment:
In your demo the font size was larger here than in the editor

Reply:
both the editor and the terminal have font size `13px`